sdk/tests/types/address/restricted.rs

```diff
@@ -0,0 +1,184 @@
+        .with_allowed_capabilities([0])
+        .unwrap();
+    let mut capabilities = address.allowed_capabilities()[0];
+
+    assert!(!capabilities.has_capabilities(CapabilityFlag::NATIVE_TOKENS));
+    capabilities.add_capabilities(CapabilityFlag::NATIVE_TOKENS);
+    assert!(capabilities.has_capabilities(CapabilityFlag::NATIVE_TOKENS));
+
+    assert!(!capabilities.has_capabilities(CapabilityFlag::MANA));
+    capabilities.set_capabilities(CapabilityFlag::MANA | CapabilityFlag::DELEGATION_OUTPUTS);
+    assert!(capabilities.has_capabilities(CapabilityFlag::MANA));
+    assert!(capabilities.has_capabilities(CapabilityFlag::DELEGATION_OUTPUTS));
+    assert!(!capabilities.has_capabilities(CapabilityFlag::NATIVE_TOKENS));
+}
+
+#[test]
+fn restricted_ed25519() {
+    // Test from https://github.com/iotaledger/tips-draft/blob/tip50/tips/TIP-0050/tip-0050.md#bech32-strings
+    let address = Ed25519Address::new(
+        Blake2b256::digest(
+            PublicKey::try_from_bytes(
+                hex::decode("6f1581709bb7b1ef030d210db18e3b0ba1c776fba65d8cdaad05415142d189f8")
+                    .unwrap()
+                    .try_into()
+                    .unwrap(),
+            )
+            .unwrap()
+            .to_bytes(),
+        )
+        .try_into()
+        .unwrap(),
+    );
+    assert_eq!(
+        hex::encode(address),
+        "efdc112efe262b304bcf379b26c31bad029f616ee3ec4aa6345a366e4c9e43a3"
+    );
+    // Ed25519 Address (Plain)
+    assert_eq!(
+        address.to_bech32_unchecked("iota"),
+        "iota1qrhacyfwlcnzkvzteumekfkrrwks98mpdm37cj4xx3drvmjvnep6xqgyzyx"
+    );
+
+    // Restricted Ed25519 Address (Every Capability Disallowed)
+    let mut address = RestrictedAddress::new(address).unwrap();
+    assert_eq!(
+        address.clone().to_bech32_unchecked("iota"),
+        "iota19qqwlhq39mlzv2esf08n0xexcvd66q5lv9hw8mz25c695dnwfj0y8gcq3l9hek"
+    );
+
+    // Restricted Ed25519 Address (Every Capability Allowed)
+    address.set_allowed_capabilities([CapabilityFlag::ALL]).unwrap();
+    assert_eq!(
+        address.clone().to_bech32_unchecked("iota"),
+        "iota19qqwlhq39mlzv2esf08n0xexcvd66q5lv9hw8mz25c695dnwfj0y8gcplupydhwt"
+    );
+
+    address.set_allowed_capabilities([CapabilityFlag::NONE]).unwrap();
+    assert_eq!(
+        address.clone().to_bech32_unchecked("iota"),
+        "iota19qqwlhq39mlzv2esf08n0xexcvd66q5lv9hw8mz25c695dnwfj0y8gcq3l9hek"
+    );
+
+    // Restricted Ed25519 Address (Can receive Native Tokens)
+    address
+        .set_allowed_capabilities([CapabilityFlag::NATIVE_TOKENS])
+        .unwrap();
+    assert_eq!(
+        address.clone().to_bech32_unchecked("iota"),
+        "iota19qqwlhq39mlzv2esf08n0xexcvd66q5lv9hw8mz25c695dnwfj0y8gcpqytmqxr4"
+    );
+
+    let address = ImplicitAccountCreationAddress::from(*address.address().as_ed25519());
+    assert_eq!(
+        address.to_bech32_unchecked("iota"),
+        "iota1rrhacyfwlcnzkvzteumekfkrrwks98mpdm37cj4xx3drvmjvnep6xg4ad2d"
```

Comment:
Where is this value from? A TIP?

Reply:
https://github.com/iotaledger/tips-draft/blob/tip50/tips/TIP-0050/tip-0050.md#bech32-strings

Reply:
it may not be there anymore, but it was when I first created the test

Reply:
Let's just remove it

Reply:
done